file/validate.go

```diff
@@ -1,13 +1,66 @@
 package file
 
 import (
+	"encoding/json"
 	"fmt"
 
 	ghodss "github.com/ghodss/yaml"
 	"github.com/kong/deck/utils"
 	"github.com/xeipuuv/gojsonschema"
 )
 
+func check(item string, t *string) bool {
+	if t != nil || len(*t) > 0 {
+		fmt.Println(item + "[err] is not allowed to be specified.")
+		return false
+	}
+	return true
+}
+
+func checkDefaults(kd KongDefaults) bool {
```

Comment:
Please change this function to return []error. In the body, please collect all the errors and return the result.